### PR TITLE
feat(editing-support): add nvim-ts-rainbow2 & add workaround for bugs 

### PR DESCRIPTION
**Pull request by utkarshgupta137**

### CODE REVIEW

File: lua/astrocommunity/editing-support/nvim-ts-rainbow2/nvim-ts-rainbow2.lua
Comment (mehalter) [inline lua/astrocommunity/editing-support/nvim-ts-rainbow2/nvim-ts-rainbow2.lua:15]:
```suggestion
return {
  "nvim-treesitter/nvim-treesitter",
  dependencies = {
    "HiPhish/nvim-ts-rainbow2",
    config = function()
      -- HACK: https://github.com/p00f/nvim-ts-rainbow/issues/112#issuecomment-1310835936
      vim.api.nvim_create_autocmd({ "BufWritePost", "FocusGained" }, {
        callback = function()
          vim.cmd "TSDisable rainbow"
          vim.cmd "TSEnable rainbow"
        end,
      })
    end,
  },
  opts = { rainbow = { enable = true } },
}
```

I think this is a better approach to adding that `autocmd`

**Review by mehalter — CHANGES_REQUESTED** [review]
@utkarshgupta137 I made a commend about how to add the hacky `autocmd` to make it better for generalizing for people using multiple packs and things. It should also be applied to you adding the hack to the `nvim-ts-rainbow`

File: lua/astrocommunity/editing-support/nvim-ts-rainbow2/nvim-ts-rainbow2.lua
Comment (utkarshgupta137) [reply lua/astrocommunity/editing-support/nvim-ts-rainbow2/nvim-ts-rainbow2.lua]: Done

File: lua/astrocommunity/editing-support/nvim-ts-rainbow/nvim-ts-rainbow.lua
Comment (mehalter) [inline lua/astrocommunity/editing-support/nvim-ts-rainbow/nvim-ts-rainbow.lua:13]:
```suggestion
```

you need to remove this line

File: lua/astrocommunity/editing-support/nvim-ts-rainbow2/nvim-ts-rainbow2.lua
Comment (mehalter) [inline lua/astrocommunity/editing-support/nvim-ts-rainbow2/nvim-ts-rainbow2.lua:13]:
```suggestion
```

This line should also be removed

**Review by mehalter — APPROVED** [review]
Looks great! Thanks so much!

**Comment by mehalter** [comment]
@utkarshgupta137 sure we can wait, I also don't mind getting it merged in initially and people can always make more PRs to improve things. Similar with improving any other plugin as well. But we can hold off on this initially

**Comment by mehalter** [comment]
@utkarshgupta137 I did some maintenance on our github actions and used this PR as a guinea pig. Just a heads up that I did rebase the PR on `main` to get the latest changes to test things

**Comment by mehalter** [comment]
@utkarshgupta137 also just to get this idea out there. I'm going to be hopefully working on some relatively restrictive github actions ci/cd status checks that will hopefully let me just enable merge queuing and automatic merging on a schedule if certain tests are passing. That way no one needs to wait for me or another repo maintainer to come and approve things. They can just meet the requirements and get things merged quickly :)

This will also stress that there isn't a "lead dev" in this repo and is mean to simply be for outside contributions

**Comment by utkarshgupta137** [comment]
A couple of ideas:
- Require at least 1 or 2 approvals (& no changes requested) before auto-merging
- Require the PR to be open for at least 24 hours before auto-merging
- If possible, only approvals by previous contributors should be considered

**Comment by szechp** [comment]
I took a look a your pr looks nice :). But ts-rainbow2 integrates with treesitter congif, so I called astrovims config to edit inbuilt ts settings . Since ts-rainbow one and two are already quite different, we should just keep both and let users decide if they want to use 1 or 2?

**Comment by mehalter** [comment]
@szechp this PR does not remove the `ts-rainbow`. They are two different plugins and therefore would be two different entries in this repository

**Comment by mehalter** [comment]
@utkarshgupta137 it's actually going to be a lot less hands on than that. Ideally we would be able to circumvent approvals in some specific scenarios:

1. require that you only add files, no removing files
1. a PR may only touch a single plugin at a time, basically one folder within a subfolder: `/lua/astrocommunity/CATEGORY/PLUGIN`
1. for a PR to qualify for automatic merging it will need to be in a category that already exists
1. Only `.lua` files and `.md` files are allowed in a PR and no nested files
1. Automatic merge queue wait time would ideally be like 24-36 hours for first time contributors and then be brought down to 60 minutes

Although I hadn't thought about just allowing approvals from anyone, that's definitely an interesting idea :) I'll think about that too while I mull over this for the next few weeks

**Pull request merged**
